storage/factory/pruningStorerFactory.go

```diff
@@ -111,121 +111,113 @@ func checkArgs(args StorageServiceFactoryArgs) error {
+		return err
 	}
+	store.AddStorer(dataRetriever.MiniBlockUnit, miniBlockUnit)
 
-	userAccountsUnit, err = psf.createTriePruningStorer(psf.generalConfig.AccountsTrieStorage, customDatabaseRemover)
+	metaBlockUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.MetaBlockStorage, disabledCustomDatabaseRemover)
+	metaBlockUnit, err := psf.createPruningPersister(metaBlockUnitArgs)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	store.AddStorer(dataRetriever.MetaBlockUnit, metaBlockUnit)
 
-	peerAccountsUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.PeerAccountsTrieStorage, customDatabaseRemover)
-	peerAccountsUnit, err = psf.createTrieUnit(psf.generalConfig.PeerAccountsTrieStorage, peerAccountsUnitArgs)
+	// metaHdrHashNonce is static
+	metaHdrHashNonceUnitConfig := GetDBFromConfig(psf.generalConfig.MetaHdrNonceHashStorage.DB)
+	dbPath := psf.pathManager.PathForStatic(shardID, psf.generalConfig.MetaHdrNonceHashStorage.DB.FilePath)
+	metaHdrHashNonceUnitConfig.FilePath = dbPath
+	metaHdrHashNonceUnit, err := storageunit.NewStorageUnitFromConf(
+		GetCacherFromConfig(psf.generalConfig.MetaHdrNonceHashStorage.Cache),
+		metaHdrHashNonceUnitConfig)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	store.AddStorer(dataRetriever.MetaHdrNonceHashDataUnit, metaHdrHashNonceUnit)
 
-	userAccountsCheckpointsUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.AccountsTrieCheckpointsStorage, disabledCustomDatabaseRemover)
-	userAccountsCheckpointsUnit, err = psf.createPruningPersister(userAccountsCheckpointsUnitArgs)
+	headerUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.BlockHeaderStorage, disabledCustomDatabaseRemover)
+	headerUnit, err := psf.createPruningPersister(headerUnitArgs)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	store.AddStorer(dataRetriever.BlockHeaderUnit, headerUnit)
 
-	peerAccountsCheckpointsUnitArgs := psf.createPruningStorerArgs(psf.generalConfig.PeerAccountsTrieCheckpointsStorage, disabledCustomDatabaseRemover)
-	peerAccountsCheckpointsUnit, err = psf.createPruningPersister(peerAccountsCheckpointsUnitArgs)
+	userAccountsUnit, err := psf.createTriePruningStorer(psf.generalConfig.AccountsTrieStorage, customDatabaseRemover)
 	if err != nil {
-		return nil, err
+		return err
 	}
+	store.AddStorer(dataRetriever.UserAccountsUnit, userAccountsUnit)
 
-	// metaHdrHashNonce is static
-	metaHdrHashNonceUnitConfig := GetDBFromConfig(psf.generalConfig.MetaHdrNonceHashStorage.DB)
-	shardID := core.GetShardIDString(psf.shardCoordinator.SelfId())
-	dbPath := psf.pathManager.PathForStatic(shardID, psf.generalConfig.MetaHdrNonceHashStorage.DB.FilePath)
-	metaHdrHashNonceUnitConfig.FilePath = dbPath
-	metaHdrHashNonceUnit, err := storageunit.NewStorageUnitFromConf(
-		GetCacherFromConfig(psf.generalConfig.MetaHdrNonceHashStorage.Cache),
-		metaHdrHashNonceUnitConfig)
+	peerAccountsUnit, err := psf.createTriePruningStorer(psf.generalConfig.PeerAccountsTrieStorage, customDatabaseRemover)
```

Comment:
This is not created correctly for shard. When creating the peersUnit for shard, it needs to have a normal persistersTracker, not a triePersisterTracker. That is because we do not snapshot the peer state for shards, so there is no need to manage the opening and closing of the dbs in a special manner.